### PR TITLE
Minimize semantic differences between library.proto and library_gapic.yaml

### PR DESCRIPTION
**Pull request by andreamlin**
This is so we can later test using only the library.proto (annotated) to generate a client, and it should have a similar surface to the client generated using the library_gapic.yaml.

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto [hunk 1]
```diff
@@ -47,7 +47,7 @@ option (google.api.metadata) = {
 // Service comment may include special characters: <>&"`'@.
 //
 service LibraryService {
-  option (google.api.default_host) = "library-example.googleapis.com";
+  option (google.api.default_host) = "library-example.googleapis.com:1234";
```
Comment (andreamlin) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto]: the service yaml at library.yaml has this as the service address

File: src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto [hunk 2]
```diff
@@ -427,8 +519,7 @@ message Shelf {
 message CreateShelfRequest {
   // The shelf to create.
   Shelf shelf = 1 [
-    (google.api.required) = true,
-    (google.api.resource_type) = "google.example.library.v1.Shelf"];
```
Comment (pongad) [inline src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto]: There's probably something I don't understand. Does the current GAPIC YAML not use resource names for these?

File: src/test/java/com/google/api/codegen/util/ProtoParserTest.java [hunk 3]
```diff
@@ -132,33 +144,34 @@ public void testIsHttpGetMethod() {
...
   public void testGetRequiredFields() {
     Method publishSeriesMethod = libraryService.lookupMethod("PublishSeries");
     List<String> requiredFields = protoParser.getRequiredFields(publishSeriesMethod);
     Collections.sort(requiredFields);
-    assertThat(requiredFields.size()).isEqualTo(2);
+    assertThat(requiredFields.size()).isEqualTo(3);
     assertThat(requiredFields.get(0)).isEqualTo("books");
-    assertThat(requiredFields.get(1)).isEqualTo("shelf");
+    assertThat(requiredFields.get(1)).isEqualTo("series_uuid");
+    assertThat(requiredFields.get(2)).isEqualTo("shelf");
```
Comment (pongad) [inline src/test/java/com/google/api/codegen/util/ProtoParserTest.java]: I think you can leave `requiredFields` unsorted then `assertThat(requiredFields).containsExactly("books", "series_uuid", "shelf");`

File: src/test/java/com/google/api/codegen/util/ProtoParserTest.java [hunk 4]
```diff
@@ -186,6 +199,12 @@ public void testGetMethodSignatures() {
     assertThat(additionalSignature.getFieldsList().size()).isEqualTo(2);
     assertThat(additionalSignature.getFieldsList().get(0)).isEqualTo("name");
     assertThat(additionalSignature.getFieldsList().get(1)).isEqualTo("message");
+
+    MethodSignature additionalSignature2 = getShelfFlattenings.get(2);
+    assertThat(additionalSignature2.getFieldsList().size()).isEqualTo(3);
+    assertThat(additionalSignature2.getFieldsList().get(0)).isEqualTo("name");
+    assertThat(additionalSignature2.getFieldsList().get(1)).isEqualTo("message");
+    assertThat(additionalSignature2.getFieldsList().get(2)).isEqualTo("string_builder");
```
Comment (pongad) [inline src/test/java/com/google/api/codegen/util/ProtoParserTest.java]: ditto

File: src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto (hunk 2 above)
Comment (andreamlin) [reply src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto]: The `resource_type` annotation only makes sense for `string` type fields. This is a weird circular reference, where the `Shelf shelf` refers to a `resource_type = "google.example.library.v1.Shelf"`, which doesn't provide any new information. I had previously put this annotation in to match what the annotated pubsub.proto used to look like, but those changes were recently reverted in https://github.com/googleapis/googleapis/pull/520

File: src/test/java/com/google/api/codegen/util/ProtoParserTest.java (hunk 3 above)
Comment (andreamlin) [reply src/test/java/com/google/api/codegen/util/ProtoParserTest.java]: neat

File: src/test/java/com/google/api/codegen/util/ProtoParserTest.java (hunk 4 above)
Comment (andreamlin) [reply src/test/java/com/google/api/codegen/util/ProtoParserTest.java]: done

**Comment by andreamlin** [comment]
PTAL
fixed tests

File: src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto (hunk 2 above)
Comment (pongad) [reply src/test/java/com/google/api/codegen/protoannotations/testdata/library.proto]: Oh right I get it now.

**Pull request merged**